test/e2e/snaps/test-snap-txinsights-v2.spec.js

```diff
@@ -37,22 +36,18 @@ describe('Test Snap TxInsights-v2', function () {
         await driver.clickElement('#connecttransaction-insights');
 
         // switch to metamask extension and click connect
-        await switchToNotificationWindow(driver);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
         await driver.clickElement({
           text: 'Connect',
           tag: 'button',
         });
 
-        await driver.waitForSelector({ text: 'Connect' });
-
         await driver.clickElement({
           text: 'Confirm',
           tag: 'button',
         });
 
-        await driver.waitForSelector({ text: 'OK' });
-
-        await driver.clickElement({
+        await driver.clickElementAndWaitForWindowToClose({
```

Comment:
this mitigates race conditions: we should not proceed to the next step until the dialog is closed